statsmodels/graphics/gofplots.py

```diff
@@ -308,10 +309,37 @@ def qqplot(self, xlabel=None, ylabel=None, line=None, other=None,
             if not check_other:
                 other = ProbPlot(other)
 
-            fig, ax = _do_plot(other.sample_quantiles,
-                               self.sample_quantiles,
-                               self.dist, ax=ax, line=line,
-                               **plotkwargs)
+            if self.nobs < other.nobs:
+                finv = interp1d([(i+1) / other.nobs for i in range(other.nobs)], other.sorted_data)
+                q = finv([(i+1) / self.nobs for i in range(self.nobs)])
+                p_plot = ProbPlot(data=q, 
+                                  dist=other.dist, fit=other.fit, 
+                                  distargs=other.distargs, a=other.a, 
+                                  loc=other.loc, scale=other.scale)
+
+                fig, ax = _do_plot(p_plot.sample_percentiles,
+                                   self.sample_percentiles,
+                                   self.dist, ax=ax, line=line,
+                                   **plotkwargs)
+
+            elif other.nobs < self.nobs:
+                finv = interp1d([(i+1) / self.nobs for i in range(self.nobs)], self.sorted_data)
+                q = finv([(i+1) / other.nobs for i in range(other.nobs)])
+                p_plot = ProbPlot(data=q, 
+                                  dist=self.dist, fit=self.fit, 
+                                  distargs=self.distargs, a=self.a, 
+                                  loc=self.loc, scale=self.scale)
+
+                fig, ax = _do_plot(other.sample_percentiles,
+                                   p_plot.sample_percentiles,
+                                   p_plot.dist, ax=ax, line=line,
+                                   **plotkwargs)
+
+            else:
+                fig, ax = _do_plot(other.sample_percentiles,
```

Comment:
Just use a single call here fo the forms

```
fig, ax = _do_plot(other_percentiles,
                                  self_percentiles,
                                   dist, ax=ax, line=line,
                                   **plotkwargs)
```

and then define `other_percentiles`, `self_percentiles` and `dist` correctly depending on the sample lengths.

Reply:
Yes, agreed.  Will work on making that change.

Since within the `if` and `elif` statements we've already computed the corresponding quantiles for the larger sample in sorted order, would you recommend passing these directly to `_do_plot` instead of initializing a new `ProbPlot` and recomputing the percentiles?

Regarding the `dist` parameter, my understanding from looking over the code is that this is a theoretical distribution which wouldn't change depending on sample size and so we could leave this alone, but please let me know if I'm mistaken on this.

I'll also put together a minimal test for the proposed change.